src/BayesFilters/src/UKFCorrection.cpp

```diff
@@ -140,5 +153,9 @@ void UKFCorrection::correctStep(const GaussianMixture& pred_state, GaussianMixtu
         /* Evaluate the filtered covariance
            P_{k}+ = P_{k}- - K * Py * K' */
         corr_state.covariance(i) = pred_state.covariance(i) - K * pred_meas.covariance(i) * K.transpose();
+
+        /* Evaluate the likelihood associated to the current component.
+           This is required to implement the method getLikelihood(). */
+        likelihood_(i) = utils::multivariate_gaussian_density(innovations.col(i), VectorXd::Zero(innovations.rows()), pred_meas.covariance(i)).coeff(0);
```

Comment:
I would rather save the `innovations` and `Py` as class data member instead of the `likelihood`. It is quite rare that you need the likelihood from the kalman filters, but is definitely more likelihood that you need the innovation and it's likelihood. If requested by the user, we can then evaluate and return the likelihood from the latest available data.

Reply:
While for class `KFCorrection` this is fine, for class `UKFCorrection` all the measurement covariances are stored in a local `GaussianMixture`. In order to have that `GaussianMixture` as a class member, we need first to close issue #48.

I suggest to wait for this issue to be closed then.

Reply:
Ping me when you push new commits and want me to review them 👍

Reply:
Pushed! :+1: Unfortunately it seems that GitHub is mixing up the order of the commits. In my local git the final commit `Implement method utils::multivariate_gaussian_density()` is right after `Implement method utils::multivariate_gaussian_log_density()` and not at the end of the list of commits :disappointed: